library/private/f3d_window_impl_standard.h

```diff
@@ -18,7 +18,14 @@ namespace detail
 class window_impl_standard : public window_impl
 {
 public:
-  window_impl_standard(const options& options, bool offscreen);
+  enum class WindowType : unsigned char
+  {
+    NATIVE,
+    NATIVE_OFFSCREEN,
+    EXTERNAL
+  };
+
+  window_impl_standard(const options& options, WindowType type);
```

Comment:
maybe use the engine flags directy ?

Reply:
I don't know, adding `#include "engine.h"` just to hack an existing enum does not look clean to me, I prefer a specific and private enum.